restli-tools/src/main/java/com/linkedin/restli/tools/compatibility/CompatibilityInfoMap.java

```diff
@@ -296,4 +303,49 @@ public boolean addAll(CompatibilityInfoMap other)
     }
     return true;
   }
+
+  public void addAnnotation(CompatibilityMessage message)
+  {
+    final CompatibilityInfo.Type infoType;
+    CompatibilityInfo info;
+    String infoMessage = String.format(message.getFormat(), message.getArgs());
+
+    if (message.isError())
+    {
+      switch (message.getImpact())
+      {
+        case BREAKS_NEW_READER:
+          infoType = CompatibilityInfo.Type.TYPE_BREAKS_NEW_READER;
+          break;
+        case BREAKS_OLD_READER:
+          infoType = CompatibilityInfo.Type.TYPE_BREAKS_OLD_READER;
+          break;
+        case BREAKS_NEW_AND_OLD_READERS:
+          infoType = CompatibilityInfo.Type.TYPE_BREAKS_NEW_AND_OLD_READERS;
+          break;
+        case ANNOTATION_INCOMPATIBLE_CHANGE:
```

Comment:
I didn't understand why there is any other error message for annotation other than `ANNOTATION_INCOMPATIBLE_CHANGE` ?

Reply:
Since we do not provide annotation compatibility rules, customers provide the way how they want to check the annotations. There are some common sense rules like - change breaks new reader, change breaks old reader or change breaks new and old reader. The ANNOTATION_INCOMPATIBLE_CHANGE provide an option to users to define the way how they want to define the compatibility other than above. We provide the framework to support it.

Reply:
then do you think CUSTOM_ANNOTATION_INCOMPATIBLE_CHANGE will be a better definition?

Reply:
Discussed offline, we will only have ANNOTATION_INCOMPATIBLE_CHANGE  here.